imageio/plugins/freeimage_v3.py

```diff
@@ -0,0 +1,378 @@
+
+
+class FreeimagePlugin(PluginV3):
+    @staticmethod
+    def flags(_fif: int, _io_mode: IOMode, _flags: int = 0, **kwargs) -> int:
+        if _fif == 0:  # BMP
+            if _io_mode == IOMode.write:
+                _flags |= IO_FLAGS.BMP_SAVE_RLE if kwargs.get('compression', False) else IO_FLAGS.BMP_DEFAULT
+        elif _fif == 1:  # ICO
+            if _io_mode == IOMode.read:
+                _flags |= IO_FLAGS.ICO_MAKEALPHA if kwargs.get('makealpha', True) else 0
+        elif _fif == 2:  # JPG
+            if _io_mode == IOMode.read:
+                _flags |= IO_FLAGS.JPEG_EXIFROTATE if kwargs.get('exifrotate', True) and kwargs.get('exifrotate', True) != 2 else 0
+                _flags |= IO_FLAGS.JPEG_ACCURATE if not kwargs.get('quickread', False) else 0
+            else:
+                quality = int(kwargs.get('quality', 75))
+                # Test quality
+                if quality < 1 or quality > 100:
+                    raise ValueError("JPEG quality should be between 1 and 100.")
+                # Build flags from kwargs
+                _flags |= quality
+                _flags |= IO_FLAGS.JPEG_PROGRESSIVE if kwargs.get('progressive', False) else 0
+                _flags |= IO_FLAGS.JPEG_OPTIMIZE if kwargs.get('optimize', False) else 0
+                _flags |= IO_FLAGS.JPEG_BASELINE if kwargs.get('baseline', False) else 0
+        elif _fif == 13:  # PNG
+            if _io_mode == IOMode.read:
+                _flags |= IO_FLAGS.PNG_IGNOREGAMMA if kwargs.get('ignoregamma', True) else 0
+            else:
+                _flags |= IO_FLAGS.PNG_INTERLACED if kwargs.get('interlaced', False) else 0
+                compression_map = {
+                    0: IO_FLAGS.PNG_Z_NO_COMPRESSION,
+                    1: IO_FLAGS.PNG_Z_BEST_SPEED,
+                    6: IO_FLAGS.PNG_Z_DEFAULT_COMPRESSION,
+                    9: IO_FLAGS.PNG_Z_BEST_COMPRESSION,
+                }
+                try:
+                    _flags |= compression_map[kwargs.get('compression', 9)]
+                except KeyError:
+                    raise ValueError("Png compression must be 0, 1, 6, or 9.")
+        elif _fif == 25:  # GIF
+            if _io_mode == IOMode.write:
+                _flags |= IO_FLAGS.GIF_PLAYBACK if kwargs.get('playback', True) else 0
+        elif _fif in [7, 8, 11, 12, 14, 15]:  # PBM, PGM, PPM
+            if _io_mode == IOMode.write:
+                _flags |= IO_FLAGS.PNM_SAVE_ASCII if kwargs.get('use_ascii', True) else 0
+
+        return _flags
+
+    def __init__(self, request: Request, flags: int = 0, **kwargs) -> None:
+        super().__init__(request)
+        self._bm, self._io_mode = None, request.mode.io_mode
+
+        _fif = -1
+        if fi.has_lib():
+            try:
+                _fif = fi.getFIF(request.filename, self._io_mode.value, request.firstbytes if self._io_mode == IOMode.read else None)
+            except Exception as e:
+                _fif = -1
```

Comment:
I remember that when _fif = -1 means FreeImage can not recognize the image format. Because after this Exception some procedures should do. I'll double check the v2 code to corfirm it.